src/controllers/ContentController.ts

```diff
@@ -296,4 +297,28 @@ export class ContentController
             this.children.delete(id);
         }
     }
+
+    protected onChildRemoved(child: DisplayObject)
+    {
+        const registeredChild = this.getChild(child);
+
+        if (registeredChild)
+        {
+            this.children.delete(registeredChild);
+            this.layout.update();
+        }
+    }
+
+    protected getChild(childInstance: DisplayObject): string | undefined
+    {
+        for (const [key, value] of this.children.entries())
+        {
+            if (value === childInstance)
+            {
+                return key;
+            }
+        }
+
+        return undefined;
+    }
```

Comment:
Just a bit more efficient way of searching through the Map, as we do not go further than needed.